src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -35,10 +35,12 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                 }
 
                 var token = root.FindToken(context.Span.Start);
-                if (IsCloseParenOrComma(token))
+                if (token.Span.Start == context.Span.Start &&
+                    IsCloseParenOrComma(token))
                 {
-                    if (token.Span.Start != context.Span.Start ||
-                        IsCloseParenOrComma(token = token.GetPreviousToken()))
+                    token = token.GetPreviousToken();
+                    if (token.Span.End != context.Span.Start || 
+                        IsCloseParenOrComma(token))
```

Comment:
yeah I did suggest that earlier.. so I need to fork FirstAncestorOrSelf method somewhere else since there is no helper for that case, I guess.